pkg/http_endpoint/v1/router.go

```diff
@@ -56,26 +60,237 @@ func authRouter(clients *common.ClientContainer,
 	return router
 }
 
-func adminRouter(authMiddleware func(http.Handler) http.Handler, secret string) *chi.Mux {
-	router := chi.NewRouter()
-	router.Use(authMiddleware)
-	router.Use(httpAuth.AdminAuthenticationMiddleware(secret))
-	/*
-		router.Get("/", func(w http.ResponseWriter, r *http.Request) {
-			_, err := w.Write([]byte("You are admin\n"))
-			if err != nil {
-			}
-		})
-	*/
-	return router
+func GetDatasource(w http.ResponseWriter, r *http.Request) {
+	dslist, err := grafanaSession.GetDataSourceList()
+	if err != nil {
+		w.Write([]byte("Failed getting datasources"))
+	} else{
+		datasources, _ := json.Marshal(dslist)
+		w.Write(datasources)
+	}
+}
+
+func GetDatasourceID(w http.ResponseWriter, r *http.Request) {
+	datasourceID := chi.URLParam(r, "datasourceID")
+	ID, _ := strconv.Atoi(datasourceID)
+	dslist, err := grafanaSession.GetDataSourceListID(ID)
```

Comment:
grafanaSession is not in context of this function, that code would not work,
to fix the issue -> move grafanaclient inside clientsContainer and wrap handler with additional function that holds context
```
func GetDatasourceID(clients *common.ClientsContainer) func(http.ResponseWriter, *http.Request) {
return func(w http.ResponseWriter, r *http.Request){
    datasourceID := chi.URLParam(r, "datasourceID")
	ID, _ := strconv.Atoi(datasourceID)
	dslist, err := clients.grafanaSession.GetDataSourceListID(ID)
	if err != nil {
￼ 		w.Write([]byte(fmt.Sprintf("Failed getting datasource by ID: %v", datasourceID)))
￼ 	} else{
￼ 		datasources, _ := json.Marshal(dslist)
￼ 		w.Write(datasources)
￼ 	}
  }
}
```
In router section
`
r.Get("/:datasourceID", GetDatasourceID(clients))
`